libLanlGeoMag/Lgm/Lgm_IGRF.h

```diff
@@ -16,27 +16,27 @@
  *
  *
  *
- *   Coefficients from: http://www.ngdc.noaa.gov/IAGA/vmod/igrf12coeffs.txt
- *   ( IGRF 12th generation (revised 2015) )
+ *   Coefficients from: https://www.ngdc.noaa.gov/IAGA/vmod/coeffs/igrf14coeffs.txt
+ *   ( IGRF 14th generation (finalized November 2024) )
  *
  */
 #ifndef M_SQRT_3
 #define M_SQRT_3        1.7320508075688772935274463415058       /* sqrt(3)        */
 #endif
 
-static int    IGRF_nModels = 25;
+static int    IGRF_nModels = 26;
 
-static double IGRF_epoch[26] = { 1900.0, 1905.0, 1910.0, 1915.0, 1920.0, 1925.0, 1930.0, 1935.0, 1940.0, 1945.0, 1950.0,
-                                1955.0, 1960.0, 1965.0, 1970.0, 1975.0, 1980.0, 1985.0, 1990.0, 1995.0, 2000.0, 2005.0, 2010.0, 2015.0, 2020.0 };
+static double IGRF_epoch[27] = { 1900.0, 1905.0, 1910.0, 1915.0, 1920.0, 1925.0, 1930.0, 1935.0, 1940.0, 1945.0,
+                                1950.0, 1955.0, 1960.0, 1965.0, 1970.0, 1975.0, 1980.0, 1985.0, 1990.0, 1995.0, 2000.0, 2005.0, 2010.0, 2015.0, 2020.0, 2025.0 };
 
-static int   IGRF_MaxN[26]  = { 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 13, 13, 13, 13 };
+static int   IGRF_MaxN[27]  = { 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 13, 13, 13 };
```

Comment:
This is deep in "I don't understand how this works" territory, but it looks like it would affect the 2015 value, changing from 13 to 0

Reply:
Good eyes! This would be a problem if this variable were actually used anywhere. It's not, although I suppose someone somewhere could've written code that relied on it...
It's nominally there to say what order the model is at each epoch, because it changed from 10 to 13 for epoch 2000 onward. The coefficient files now include zero for the higher order coefficients pre-2000, so there's nothing to keep track of. This is a legacy inclusion from when it mattered (the convert script is not new, see other comment).

What do you think: remove the line in the script and header?

Reply:
Yeah, I'd remove it...if someone is relying on it, it's wrong (0) for newer epochs anyhow.

Reply:
Removed in `fmt_igrf.c` and Lgm_IGRGF.h regenerated.